### PR TITLE
add <= support to loop unroller

### PR DESCRIPTION
**Pull request by WalterBright**
This enables loop unrolling for loops like:
```
void bar(int);

void something(int y)
{
    for (int i = 0; i <= 1; ++i)   // enable this style
//    for (int i = 0; i < 2; ++i)  // rather than just this style
        bar(i);
}
```
with the `<=` loop condition.

Partial solution for https://issues.dlang.org/show_bug.cgi?id=20991

### CODE REVIEW

File: src/dmd/backend/gloop.d [hunk 1]
```diff
@@ -3704,6 +3704,7 @@ bool loopunroll(ref loop l)
     if (l.Lhead.Bflags & BFLnounroll)
         return false;
     l.Lhead.Bflags |= BFLnounroll;
+    //WRfunc();
```
Comment (UplinkCoder) [inline src/dmd/backend/gloop.d:3707]: what's `WRfunc` ?

File: test/runnable/mars1.d [hunk 2]
```diff
@@ -1976,6 +1976,24 @@ void testCpStatic()
...
+void bar7(int i)
+{
+    assert(i == x7);
+    ++x7;
+}
+
+void test7()
+{
+    for (int i = 0; i <= 1; ++i)
+	bar7(i);
```
Comment (ibuclaw) [inline test/runnable/mars1.d:1993]: Is this a tab?

File: src/dmd/backend/gloop.d [hunk 3]
```diff
@@ -3734,13 +3735,13 @@ bool loopunroll(ref loop l)
...
     {
         printf("Unroll candidate:\n");
-        printf("  head:\t"); WReqn(l.Lhead.Belem); printf("\n");
-        printf("  tail:\t"); WReqn(l.Ltail.Belem); printf("\n");
+        printf("  head B%d:\t", l.Lhead.Bdfoidx); WReqn(l.Lhead.Belem); printf("\n");
+        printf("  tail B%d:\t", l.Ltail.Bdfoidx); WReqn(l.Ltail.Belem); printf("\n");
     }
 
-    /* Tail must be of the form: (v < c) where v is an unsigned integer
+    /* Tail must be of the form: (v < c) or (v <-= c) where v is an unsigned integer
```
Comment (dnadlinger) [inline src/dmd/backend/gloop.d:3742]:
```suggestion
    /* Tail must be of the form: (v < c) or (v <= c) where v is an unsigned integer
```

**Review by dnadlinger — APPROVED** [review]
In LDC, we'd add a codegen output test for things like this, but I guess it's par for the course for the DMD backend.

**Review by andralex — APPROVED** [review]

**Comment by 12345swordy** [comment]
3 days is up.

**Review by MoonlightSentinel — APPROVED** [review]

**Pull request merged**